fitbenchmarking/parsing/fitbenchmark_parser.py

```diff
@@ -29,35 +45,64 @@ def parse(self):
         """
         fitting_problem = FittingProblem()
 
-        self._entries = self._get_fitbenchmark_data_problem_entries()
+        self._entries = self._get_data_problem_entries()
+        software = self._entries['software'].lower()
+        if (software not in import_success) or (not import_success[software]):
+            raise ImportError('Could not import necessary modules for software'
+                              ' ({})'.format(software))
+
         self._parsed_func = self._parse_function()
 
+        # NAME
         fitting_problem.name = self._entries['name']
 
-        data_points = self._get_data_points()
+        # DATA
+        if software == 'mantid':
+            data_points = self._get_mantid_data_points()
```

Comment:
Can we merge the `get_data_points` for both mantid and sasview?

Reply:
done